### ISSUE TITLE
python plugin lib removes line breaks from doc string for help function

### ISSUE THREAD

**Issue by renepickhardt** [ElementsProject/lightning#2216]
As mentioned in #2139 the line breaks are removed from the doc string of plugin functions when the doc string is depicted as the help string. I thought this was a mistake and wanted to fix it. However looking at the code it looks desperately: 

https://github.com/ElementsProject/lightning/blob/0321d79540e4d2ef5296c4717cae390321b3b7d5/contrib/pylightning/lightning/plugin.py#L245

Is the rational behind this to keep the API doc clean? For the funds plugin it would be much nicer if the list with all the line breaks would stay like that list. 

I opened an issue instead of a PR since this line seems very desperately put there.

**Comment by cdecker** [comment]
The rationale for this change was that we use this help message in the command line result for `lightning-cli help`, which gets escaped and thus we end up printing a couple of `\n` in the help message. A more durable and clean change could be to keep the newlines and then do some post-processing in the lightning-cli instead when using human readable format (split lines and indent each one so it is displayed correctly).

**Comment by renepickhardt** [comment]
I guess you are talking about the code block following this line?
https://github.com/ElementsProject/lightning/blob/master/cli/lightning-cli.c#L56

why is this replace actually being made in the first place?

**Comment by cdecker** [comment]
That's in order to make the human readable output more readable by indenting the help text below the command

**Comment by NishantBansal2003** [comment]
Hi @cdecker,  
I was working on this issue and noticed that the `doc` variable is not being used anywhere: https://github.com/ElementsProject/lightning/blob/2f6428394501fe19a2228271f548f6d5540aa85d/contrib/pyln-client/pyln/client/plugin.py#L941-L961.  
Additionally, when I use the plugin and run the `lightning-cli help` command, I get the following output for the `funds` plugin:  
```bash
funds [unit] \\nLists the total funds the lightning node owns off- and on-chain in {unit}.\\n{unit} can take the following values:\\ns, satoshi, satoshis to represent satoshis\\nb, bit, bits to represent bits\\nm, milli, btc to represent milliBitcoin\\nB, bitcoin, btc to represent Bitcoins\\nWhen not using Satoshis (default), the values are rounded off.
```
Would it make sense to open a separate issue to address both the unused `doc` variable and the formatting issues in the help output?

**Comment by cdecker** [comment]
Yeah, let's create a new issue. The usage of the doc string has changed substantially over the last years, and I'm not sure I know how it's used at the moment.
